src/standalone/Bucket/Bucket.jl

```diff
@@ -576,6 +591,76 @@ function surface_air_density(
     return p.bucket.ρ_sfc
 end
 
+"""
+
+    ClimaLand.total_energy_per_area!(
+        surface_field,
+        model::BucketModel,
+        Y,
+        p,
+        t,
+    )
+
+A function which updates `surface_field` in place with the value for
+the total energy per unit ground area for the `BucketModel`.
+
+The ground of the bucket model has temperature `Y.bucket.T`, with
+volumetric specific heat approximated with the parameter `ρc_soil`.
+Additional energy is present due to the latent heat of fusion of
+frozen water, in the form of snow. We also add in this energy (below).
+We do not model or account for the sensible energy of snow (`ρ_snow T_snow`),
+as this is much smaller.
+"""
+function ClimaLand.total_energy_per_area!(
+    surface_field,
+    model::BucketModel,
+    Y,
+    p,
+    t,
+)
+    surface_field .= 0
+    # Sensible energy of the soil
+    ClimaCore.Operators.column_integral_definite!(
+        surface_field,
+        model.parameters.ρc_soil .* Y.bucket.T,
+    )
+
+    # Latent heat of fusion of frozen water in the snow
+    surface_field .+=
+        -LP.LH_f0(model.parameters.earth_param_set) .*
+        LP.ρ_cloud_liq(model.parameters.earth_param_set) .* Y.bucket.σS
+    return nothing
+end
+
+"""
+
+    ClimaLand.total_liq_water_vol_per_area!(
+        surface_field,
+        model::BucketModel,
+        Y,
+        p,
+        t,
+    )
+
+A function which updates `surface_field` in place with the value for
+the total water per unit ground area for the `BucketModel`, in kg m⁻².
+
+The total water contained in the bucket is the sum of the
+subsurface water storage `W`, the snow water equivalent `σS` and
+surface water content `Ws`, times the density of liquid water `ρ_cloud_liq`.
+"""
+function ClimaLand.total_liq_water_vol_per_area!(
+    surface_field,
+    model::BucketModel,
+    Y,
+    p,
+    t,
+)
+    ρ_cloud_liq = LP.ρ_cloud_liq(model.parameters.earth_param_set)
+    @. surface_field = (Y.bucket.σS + Y.bucket.W + Y.bucket.Ws) * ρ_cloud_liq
```

Comment:
This is converting from a liquid volume per area to a mass per area. I think given the function name this should be handled in the coupler...